src/base/bittorrent/torrentinfo.cpp

```diff
@@ -76,12 +76,23 @@ namespace
 const int torrentInfoId = qRegisterMetaType<TorrentInfo>();
 
 TorrentInfo::TorrentInfo(std::shared_ptr<const lt::torrent_info> nativeInfo)
+    : m_nativeInfo {std::const_pointer_cast<lt::torrent_info>(nativeInfo)}
 {
-    m_nativeInfo = std::const_pointer_cast<lt::torrent_info>(nativeInfo);
+    if (!m_nativeInfo)
+        return;
+
+    const lt::file_storage &fileStorage = m_nativeInfo->files();
+    m_nativeIndexes.reserve(fileStorage.num_files());
+    for (const lt::file_index_t nativeIndex : fileStorage.file_range())
+    {
+        if (!fileStorage.pad_file_at(nativeIndex))
+            m_nativeIndexes.append(nativeIndex);
```

Comment:
seems `m_nativeIndexes` is storing non-padding file's index. maybe rename it to `m_nonPaddingFilesIndexes`?

Reply:
I wouldn't look at it that way. These are just the internal indexes of all those files that we are dealing with. I would avoid mentioning `.pad` files wherever their mention is not explicitly required.

Reply:
OK maybe add a bit comment at the declaration because I don't really understand its purpose on sight.

Reply:
>OK maybe add a bit comment at the declaration because I don't really understand its purpose on sight and the naming is a bit generic that doesn't help understanding.

Done.